esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -3721,6 +3722,51 @@ void CWsWuFileHelper::createWUZAPFile(IEspContext& context, IConstWorkUnit* cwu,
         throw MakeStringException(ECLWATCH_CANNOT_COMPRESS_DATA,"Failed to execute system command 'zip'. Please make sure that zip utility is installed.");
 }
 
+void CWsWuFileHelper::createZAPWUQueryAssociatedFiles(IConstWorkUnit* cwu, const char* pathToCreate)
+{
+    Owned<IConstWUQuery> query = cwu->getQuery();
+    if (!query)
+    {
+        IERRLOG("Cannot get Query for workunit %s.", cwu->queryWuid());
+        return;
+    }
+
+    Owned<IConstWUAssociatedFileIterator> iter = &query->getAssociatedFiles();
+    ForEach(*iter)
+    {
+        SCMStringBuffer name, ip;
+        IConstWUAssociatedFile& cur = iter->query();
+        cur.getName(name);
+        cur.getIp(ip);
+
+        RemoteFilename rfn;
+        SocketEndpoint ep(ip.str());
+        rfn.setPath(ep, name.str());
+
+        OwnedIFile sourceFile = createIFile(rfn);
+        if (!sourceFile)
+        {
+            IERRLOG("Cannot open %s on %s.", name.str(), ip.str());
+            continue;
+        }
+
+        StringBuffer fileName(name.str());
+        getFileNameOnly(fileName, false);
+
+        StringBuffer outFileName(pathToCreate);
+        outFileName.append(PATHSEPCHAR).append(fileName);
```

Comment:
What happens here is pathToCreate is PATHSEPCHAR ("\")?  Will you end up with "\\" ?

Reply:
The pathToCreate is set in the createWorkingFolder(). It is 'tempzipfiles/ZAPReport_wuid_user'.